File: coding/Leetcode/Tasks/task_2348.cpp
// https://leetcode.com/problems/number-of-zero-filled-subarrays/description/
// Created by ruslan515 on 22.03.23.
//

#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    long long zeroFilledSubarray(vector<int>& nums) {
        int n = nums.size();
        int i = 0;
        long long answer = 0;
        while (i < n) {
            while (i < n && nums[i] != 0) {
                ++i;
            }
            int len = 0;
            while (i < n && nums[i] == 0) {
                ++len;
                i++;
            }
            long long  count;
            for (int x = 1; x <= len; ++x) {
                count = len - x + 1;
                answer += count;
            }

        }

        return answer;
    }
};

void read_data(vector<int>& nums) {
    int t;
    while (cin >> t)
        nums.push_back(t);
    return;
}

int main() {
    vector<int> nums;
    read_data(nums);
    Solution solve;
    cout << solve.zeroFilledSubarray(nums);
    return 0;
}

File: coding/Leetcode/Tasks/task_2348.java
package leetcode.tasks;


public class task_2348 {
    class Solution {
        public long zeroFilledSubarray(int[] nums) {
            long answer = 0;
            int n = nums.length;
            int left = 0, right = 0;
            long cnt, len;
            while (right < n) {
                while (left < n && nums[left] != 0) {
                    ++left;
                }
                if (left == n) {
                    break;
                }
                right = left;
                while (right < n && nums[right] == 0) {
                    ++right;
                }
                len = (long) right - left;
                cnt = len * (len + 1) / 2;
                answer += cnt;
                left = right;
            }

            return answer;
        }
    }

    public void main(String[] args) {
        int[] nums;
        long answer;

        Solution solution = new Solution();

        nums = new int[]{1, 3, 0, 0, 2, 0, 0, 4};
        answer = 6;
        assert answer == solution.zeroFilledSubarray(nums) : "Answer is different";

        nums = new int[]{0, 0, 0, 2, 0, 0};
        answer = 9;
        assert answer == solution.zeroFilledSubarray(nums) : "Answer is different";

        nums = new int[]{2, 10, 2019};
        answer = 0;
        assert answer == solution.zeroFilledSubarray(nums) : "Answer is different";
    }

}
